File: BlogServer/Blog.Service/Api/RandomWriteService.cs

using Blog.Model.Entity;
using Blog.Model.Param;
using Blog.Model.Rsult;
using Blog.Service.Api.Interface;
using Newtonsoft.Json;
using SqlSugar;

namespace Blog.Service.Api
{
    public class RandomWriteService: IRandomWriteService
    {
        public RandomWriteService(ISqlSugarClient client, IFileService fileService)
        {
            Db = client;
            FileService = fileService;
        }

        public ISqlSugarClient Db { get; }
        public IFileService FileService { get; }

        public async Task Insert(RandomWriteInsertParam param)
        {
            //var user = await Db.Queryable<UserEnity>()
            //    .Where(it => it.Id == param.CreateUserId)
            //    .FirstAsync();
            var randomWrite = new RandomWriteEnity
            {   
                Clicks = 0,
                CreateDate = DateTime.Now,
                CreateUserId = param.CreateUserId,
                Content = param.Content,
                Cover_list = param.Cover_list != null ? JsonConvert.SerializeObject(param.Cover_list) : null
            };
            await Db.Storageable(randomWrite).ExecuteCommandAsync();
        }

        public async Task Update(RandomWriteUpdateParam param)
        {
            var randomWrite = await Db.Queryable<RandomWriteEnity>()
                .Where(x => x.Id == param.Id)
                .FirstAsync();
            randomWrite.Content = param.Content;
            randomWrite.Cover_list = param.Cover_list != null ? JsonConvert.SerializeObject(param.Cover_list) : null;
            randomWrite.UpdateTime = DateTime.Now;
            await Db.Storageable(randomWrite).ExecuteCommandAsync();
        }

        public async Task<PageResult<RandomWriteFindRsult>> List(RandomWriteFindParam param)
        {
            RefAsync<int> total = 0;
            var list = await Db.Queryable<RandomWriteEnity>()
                .WhereIF(!string.IsNullOrEmpty(param.Content), it => it.Content!.Contains(param.Content!))
                .OrderBy(it => it.CreateDate, OrderByType.Desc)
                .ToPageListAsync(param.Page, param.PageNum, total);
            return new PageResult<RandomWriteFindRsult>
            {
                List = list.Select(x => new RandomWriteFindRsult(x)).ToList(),
                Total = total
            };
        }

        public async Task<RandomWriteFindRsult> FindOne(IDParam param)
        {
            var randomWrite = await Db.Queryable<RandomWriteFindRsult>().Where(it => it.Id == param.Id).FirstAsync();
            return new RandomWriteFindRsult(randomWrite);
        }

        public async Task Delete(IDParam param)
        {
            var randomWrite = await Db.Queryable<RandomWriteEnity>().Where(it => it.Id == param.Id).FirstAsync();
            //Console.WriteLine(randomWrite.Cover_list);
            //Console.WriteLine(randomWrite.Cover_list != null);
            if (randomWrite.Cover_list != null)
            {
                //Console.WriteLine(randomWrite.Cover_list);
                var Cover_list = JsonConvert.DeserializeObject<List<string>>(randomWrite.Cover_list);
                if (Cover_list!.Count != 0) FileService.ImgsRemove(Cover_list);
            }
            await Db.Deleteable<RandomWriteEnity>().In(it => it.Id,param.Id).ExecuteCommandAsync()  ;
        }
    }
}

File: blog-server/src/module/admin/randomWrite/RandomWrite.service.ts
import { Injectable } from '@nestjs/common'
import { InjectRepository } from '@nestjs/typeorm'
import { Like, Repository } from 'typeorm'
import { RandomWrite } from '../../../db/entities/RandomWrite'
import { FileService } from '../file/file.service'
import { assignProps, formatTime } from '../../../common/commonly.fun'
import {RandomWriteFindParam, RandomWriteInsertParam, RandomWriteUpdateParam} from "./model/RandomWrite.param";
import {PageRsult} from "../../../model/BaseRsult";
import {IdParam} from "../../../model/BaseParam";


@Injectable()
export class RandomWriteService {
    constructor(
        @InjectRepository(RandomWrite) private randomWriteRepository: Repository<RandomWrite>,
        private fileService: FileService
    ) {}

    // 新增
    async insert(body: RandomWriteInsertParam): Promise<void> {
        const random = new RandomWrite()
        assignProps(random, body)
        // random.createUserId = body.CreateUserId
        random.createDate = formatTime()
        // random.content = body.content
        random.clicks = 0
        if (body.cover_list) random.cover_list = JSON.stringify(body.cover_list)
        await this.randomWriteRepository.save(random)
    }


    // 修改
    async update(body:RandomWriteUpdateParam): Promise<void> {
        const random = await this.randomWriteRepository.findOne({where: {id: body.id}})
        random.updateTime = formatTime()
        random.content = body.content
        if (body.cover_list) random.cover_list = JSON.stringify(body.cover_list)
        await this.randomWriteRepository.save(random)
    }

    // 查询
    async list(body: RandomWriteFindParam): Promise<PageRsult<RandomWrite>> {
        const list = await this.randomWriteRepository.find({
            where: { content: Like(`%${ body.content || '' }%`) },
            order: { createDate: "DESC" },
            skip: body.pageNum * ( body.page - 1),
            take: body.pageNum
        })
        list.forEach((element, index) => {
            if (list[index].cover_list) {
                list[index].cover_list = JSON.parse(list[index].cover_list)
            }
        })
        const total = await this.randomWriteRepository.count({where: { content: Like(`%${ body.content || '' }%`) }})
        return {
            list,
            total
        }
    }



    // 单个查询
    async findOne(body: IdParam): Promise<RandomWrite> {
        const result = await this.randomWriteRepository.findOne({
            where: { id: body.id }
        })
        if (result.cover_list) result.cover_list = JSON.parse(result.cover_list)
        return result
    }


    // 删除
    async delete(body:IdParam) :Promise<void> {
        const result = await this.randomWriteRepository.findOne({ where: { id: body.id } })
        if (result.cover_list && result.cover_list.length) await this.fileService.imgRemove(JSON.parse(result.cover_list))
        await this.randomWriteRepository.delete({ id: body.id })
    }

}